include/fmt/core.h

```diff
@@ -329,7 +329,14 @@ template <typename... T> FMT_CONSTEXPR void ignore_unused(const T&...) {}
 
 constexpr FMT_INLINE auto is_constant_evaluated(
     bool default_value = false) noexcept -> bool {
-#ifdef __cpp_lib_is_constant_evaluated
+// Workaround for incompatible between libstdc++ consteval based
+// std::is_constant_evaluated implementation and clang-14+.
+// https://github.com/fmtlib/fmt/issues/3247
+#if FMT_CPLUSPLUS >= 202002L && defined(_GLIBCXX_RELEASE) && \
+    _GLIBCXX_RELEASE >= 12 && FMT_CLANG_VERSION >= 14
```

Comment:
Should this be `FMT_CLANG_VERSION == 14` per https://github.com/fmtlib/fmt/issues/3247#issuecomment-1399299689?

Reply:
Should be
```
(FMT_CLANG_VERSION >= 1400 && FMT_CLANG_VERSION < 1500)
```
This is my mistake, I forgot about the minor version part (00):

https://github.com/fmtlib/fmt/blob/f89cd276f7dead38f11cebc73d1e91a1b1b38124/include/fmt/core.h#L23

Since bug is fixed in clang-15, I added a second condition.